pinot-core/src/main/java/com/linkedin/pinot/core/segment/DefaultSegmentNameGenerator.java

```diff
@@ -41,11 +48,12 @@ public DefaultSegmentNameGenerator(final String segmentName) {
    * @param tableName
    * @param segmentNamePostfix
    */
-  public DefaultSegmentNameGenerator(String timeColumnName, String tableName, String segmentNamePostfix) {
+  public DefaultSegmentNameGenerator(String timeColumnName, String tableName, String segmentNamePostfix, int sequenceId) {
     _timeColumnName = timeColumnName;
     _tableName = tableName;
     _segmentNamePostfix = segmentNamePostfix;
     _segmentName = null;
+    _sequenceId = sequenceId;
   }
 
   @Override
```

Comment:
Add a javadoc here giving examples of what the segment name would look like with various combinations.